stacks-signer/src/v0/signer.rs

```diff
@@ -0,0 +1,318 @@
+                debug!("{self}: Receved a new burn block event for block height {height}")
+            }
+            None => {
+                // No event. Do nothing.
+                debug!("{self}: No event received")
+            }
+        }
+    }
+
+    fn process_command(
+        &mut self,
+        _stacks_client: &StacksClient,
+        _current_reward_cycle: u64,
+        command: Option<RunLoopCommand>,
+    ) {
+        if let Some(command) = command {
+            warn!("{self}: Received a command: {command:?}. V0 Signers do not support commands. Ignoring...")
+        }
+    }
+}
+
+impl From<SignerConfig> for Signer {
+    fn from(signer_config: SignerConfig) -> Self {
+        let stackerdb = StackerDB::from(&signer_config);
+        debug!(
+            "Reward cycle #{} Signer #{}",
+            signer_config.reward_cycle, signer_config.signer_id,
+        );
+        let signer_db =
+            SignerDb::new(&signer_config.db_path).expect("Failed to connect to signer Db");
+
+        Self {
+            private_key: signer_config.stacks_private_key,
+            stackerdb,
+            mainnet: signer_config.mainnet,
+            signer_id: signer_config.signer_id,
+            signer_addresses: signer_config
+                .signer_entries
+                .signer_ids
+                .into_keys()
+                .collect(),
+            signer_slot_ids: signer_config.signer_slot_ids.clone(),
+            reward_cycle: signer_config.reward_cycle,
+            signer_db,
+        }
+    }
+}
+
+impl Signer {
+    /// Determine this signers response to a proposed block
+    /// Returns a BlockResponse if we have already validated the block
+    /// Returns None otherwise
+    fn determine_response(&self, block_info: &mut BlockInfo) -> Option<BlockResponse> {
+        let valid = block_info.valid?;
+        let response = if valid {
+            debug!("{self}: Accepting block {}", block_info.block.block_id());
+            let signature = self
+                .private_key
+                .sign(&block_info.signer_signature_hash().0)
+                .expect("Failed to sign block");
+            BlockResponse::accepted(block_info.signer_signature_hash(), signature)
+        } else {
+            debug!("{self}: Rejecting block {}", block_info.block.block_id());
+            BlockResponse::rejected(
+                block_info.signer_signature_hash(),
+                RejectCode::RejectedInPriorRound,
+            )
+        };
+        Some(response)
+    }
+
+    /// Handle block proposal messages submitted to signers stackerdb
+    fn handle_block_proposal(
+        &mut self,
+        stacks_client: &StacksClient,
+        block_proposal: &BlockProposal,
+    ) {
+        debug!("{self}: Received a block proposal: {block_proposal:?}");
+        if block_proposal.reward_cycle != self.reward_cycle {
+            // We are not signing for this reward cycle. Reject the block
```

Comment:
Again, I'm not sure this is the case.  The prerequisite for signing is that you know whether or not your key is in the active reward cycle's signing set.  You don't even need to know how many PoX reward slots you have.  In fact, you don't even need to worry about whether or not you have zero, since if you have zero, you won't be able to post your signature to the StackerDB anyway.

This is important because on a reward cycle boundary, this condition will be true for some signers and false for others (as their nodes process the first block in the next reward cycle).  I think this check should be removed.

Reply:
I do not think this is true. a miner proposes a block for a very specific reward cycle (this is not debatable). if you are not a signer in that reward cycle which you for sure ar enot because we have a signer instantiated per reward cycle, you should not sign the block because your signature is not valid anyway so its a waste.

Reply:
I think this check should remain in place, but the `warn!` should be downgraded to `debug!` and the comment updated from `Reject the block` to `Ignore the block.`

Reply:
Will do!